api/Port.go

```diff
@@ -276,3 +286,118 @@ func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates
+	if err != nil {
+		return err
+	}
+	// interface table ops
+	intfOp = libovsdb.Operation{
+		Op:       "insert",
+		Table:    interfaceTable,
+		Row:      intf,
+		UUIDName: namedIntfUUID,
+	}
+
+	// 2) Insert a row for Nuage port in OVSDB Port table
+	portOp := libovsdb.Operation{}
+	port := make(map[string]interface{})
+	port["name"] = intfName
+	port["interfaces"] = libovsdb.UUID{namedIntfUUID}
+	port["external_ids"], err = libovsdb.NewOvsMap(extIDMap)
+	if err != nil {
+		return err
+	}
+	portOp = libovsdb.Operation{
+		Op:       "insert",
+		Table:    portTable,
+		Row:      port,
+		UUIDName: namedPortUUID,
+	}
+
+	// 3) Mutate the Ports column of the row in the Bridge table with new Nuage port
+	mutateUUID := []libovsdb.UUID{libovsdb.UUID{namedPortUUID}}
+	mutateSet, _ := libovsdb.NewOvsSet(mutateUUID)
+	mutation := libovsdb.NewMutation("ports", "insert", mutateSet)
+	condition := libovsdb.NewCondition("name", "==", bridgeName)
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     bridgeTable,
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	operations := []libovsdb.Operation{intfOp, portOp, mutateOp}
+	reply, err := vrsConnection.ovsdbClient.Transact(OvsDBName, operations...)
+	if err != nil || len(reply) < len(operations) {
+		return fmt.Errorf("Problem mutating row in the OVSDB Bridge table for alubr0")
+	}
+
+	return nil
+}
+
+// RemovePortFromAlubr0 will remove a port from alubr0 bridge
+func (vrsConnection *VRSConnection) RemovePortFromAlubr0(portName string) error {
+
+	condition := libovsdb.NewCondition("name", "==", portName)
+	selectOp := libovsdb.Operation{
+		Op:    "select",
+		Table: "Port",
+		Where: []interface{}{condition},
+	}
+
+	selectOperation := []libovsdb.Operation{selectOp}
+	reply, err := vrsConnection.ovsdbClient.Transact(OvsDBName, selectOperation...)
+	if err != nil {
+		return fmt.Errorf("Problem selecting row in the OVSDB Port table for alubr0")
+	}
+
+	// Obtain Port table OVSDB row corresponding to the port name
+	ovsdbRow := reply[0].Rows[0]
+	portUUID := ovsdbRow["_uuid"]
+	portUUIDStr := fmt.Sprintf("%v", portUUID)
+	portUUIDNew := util.SplitUUIDString(portUUIDStr)
+
+	condition = libovsdb.NewCondition("name", "==", portName)
+	deleteOp := libovsdb.Operation{
+		Op:    "delete",
+		Table: "Port",
+		Where: []interface{}{condition},
+	}
+
+	// Deleting a Bridge row in Bridge table requires mutating the open_vswitch table.
+	mutateUUID := []libovsdb.UUID{libovsdb.UUID{portUUIDNew}}
+	mutateSet, _ := libovsdb.NewOvsSet(mutateUUID)
+	mutation := libovsdb.NewMutation("ports", "delete", mutateSet)
+	condition = libovsdb.NewCondition("name", "==", bridgeName)
+
+	// simple mutate operation
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     "Bridge",
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	operations := []libovsdb.Operation{deleteOp, mutateOp}
```

Comment:
@vareti We do not need to explicitly make an OVSDB operation to remove entry from the interface table. When a port entry is deleted from the port table, OVS automatically removes the interfaces that port is referencing from the Interface table automagically. This is the reason we only cleanup from Port and Bridge table. I have tested this behaviour using our SDK tests and see the interface table entries getting cleaned up after entry gets removed from Port table.